src/presentational/AssignmentForm.jsx

```diff
@@ -0,0 +1,149 @@
+import { Component, PropTypes } from 'react';
+import { Link } from 'react-router';
+
+import InputElement from './InputElement';
+
+const initialState = {
+  classifications_target: '',
+  description: '',
+  duedate: '',
+  name: '',
+  students: [],
+  subjects: [],
+}
+
+
+class AssignmentForm extends Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.renderStudentList = this.renderStudentList.bind(this);
+    this.toggleStudent = this.toggleStudent.bind(this);
+    this.state = Object.assign({}, initialState);
+  }
+
+  toggleStudent(id) {
+    const { students } = this.state;
+    if (students.includes(id)) {
+      this.setState({
+        students: students.filter(student => student !== id),
+      })
+    } else {
+      this.setState({
+        students: students.concat(id),
+      })
+    }
+  }
+
+  handleChange(e) {
+    const nextState = {};
+    nextState[e.target.name] = e.target.value;
+    this.setState(nextState);
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const newAssignment = {};
+    for (const key in this.state) {
+      if (this.state.hasOwnProperty(key)) {
+        newAssignment[key] = this.state[key];
+      }
+    }
+    if (newAssignment.students.length > 0 && newAssignment.subjects.length) {
+      this.props.submitForm(newAssignment, this.props.params.classroomId)
+    } else {
+      alert('You can\'t create an assignment without students or subjects.')
+    };
+  }
+
+  renderStudentList(students) {
+    return (
+      <div>
+        {(students.length > 0) ?
+        <table className="table table-hover">
+          <thead>
+            <tr>
+              <th>Name</th>
+              <th>Action</th>
+            </tr>
+          </thead>
+          <tbody>
+            {students.map((student) =>
+            <tr
+              className={this.state.students.find(id => id === student.id) ? 'success' : ''}
+              key={student.id}
+            >
+              <td>
+                {student.attributes.zooniverse_display_name}
+              </td>
+              <td>
+                <input type="checkbox" onClick={this.toggleStudent.bind(this, student.id)} />
```

Comment:
I'm an idiot. My snippet above should have used `e.target.value` and `e.target.checked`.